### ISSUE TITLE
Upgrade JupyterHub to 1.3.0 and Kubernetes 1.19

### ISSUE THREAD

**Issue by zonca** [det-lab/jupyterhub-deploy-kubernetes-jetstream#48]
Zero-to-jupyterhub released 0.10 with JupyterHub 1.2.1,
we currently have 0.9 with JupyterHub 1.1.0.

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/CHANGELOG.md

In principle it should be an easy upgrade, I can do it directly in the production instance and if anything stops working, I should be able to revert to the old version.

I will do it when I see no-one is connected and I'll announce it here and tag some users.

**Comment by zonca** [comment]
@pibion I tested a newer version of the deployment, see https://zonca.dev/2021/01/kubernetes-jetstream-kubespray.html
This brings a new Kubernetes and a new JupyterHub version.
I need to redeploy from scratch and it will probably take a full day to do the update, so we need coordination.

I will try to save all the user volumes, but you never know what happens during a full redeploy, so I'd like to make sure the users also backed up everything.

**Comment by zonca** [comment]
@pibion would you like to schedule the upgrade in 1 or 2 weeks?

**Comment by pibion** [comment]
@zonca it's a good time to do maintenance.  The current user doesn't have any non-versioned secondary data, so if things are lost he can re-clone and recover everything.

**Comment by zonca** [comment]
it would be even better if you tell me the username of people I should save data for, so I can delete all the other volumes, so we can cleanup the account a bit.

here are the volumes I have:

```
00kevinmac                                                                                                                             
amilasamurdhi                                                                                                                          
bloer                                                                                                                                  
emanuele-2emichielin                                                                                                                   
enzezhang                                                                                                                              
fortuneemmanuel-2dking2023                                                                                                             
gardn701                                                                                                                               
glass-2dships                                                                                                                          
herogers924                                                                                                                            
isakia                                                                                                                                 
jpanmany466                                                                                                                            
mbaiocchi                                                                                                                              
mdlama                                                                                                                                 
nels8013                                                                                                                               
pandeylekhraj                                                                                                                          
pibion                                                                                                                                 
rac101                                                                                                                                 
radioapple                                                                                                                             
scorzasil                                                                                                                              
tosin-2dda-2dglitch                                                                                                                    
ziqinghong                                                                                                                             
zkromer                                                                                                                                
zonca 
```

**Comment by pibion** [comment]
@zonca I've been deleting users who are no longer in the collaboration and/or who know they won't be using the XSEDE JupyterHub in the future.  There are two more that I may be able to delete, I'm just waiting to confirm these.

This leaves us with still quite a few users.  Will it be troublesome to keep all these?  If so I can likely trim the list down further.  Ideally I'd like to save all these because it takes a little while to set up the ssh keys we need to access different repositories.  But it's not the end of the world if people need to re-do that.

```
ziqinghong
zonca
amilasamurdhi
bloer 
emanuele.michielin
enzezhang
gardn701
herogers924 
jpanmany466 
mbaiocchi 
nels8013 
pandeylekhraj
rac101
scorzasil
tosin-da-glitch
zkromer
```

**Comment by zonca** [comment]
no problem, I'll keep those

**Comment by zonca** [comment]
ok, I deleted the volumes of people that left the collaboration, I will keep all the others

**Comment by pibion** [comment]
@zonca there's one more we can delete: jpanmany466.

We can consider this list final:

```
ziqinghong
zonca
amilasamurdhi
bloer 
emanuele.michielin
enzezhang
gardn701
herogers924 
jpanmany466 
mbaiocchi 
nels8013 
pandeylekhraj
rac101
scorzasil
tosin-da-glitch
zkromer
```

**Comment by zonca** [comment]
ok deleted `jpanmany466`

**Comment by zonca** [comment]
@pibion @zkromerUCD I'm starting the upgrade now, I'll let you know once it is completed, the hub will be probably down all day.

**Comment by zonca** [comment]
* saved the `data` volume and the user volumes
* saved the floating IP
* deleted all openstack resource manually from Horizon

Now starting with https://zonca.dev/2021/01/kubernetes-jetstream-kubespray.html

**Comment by zonca** [comment]
ok @pibion @zkromerUCD, it is back up, it seems I managed to transfer all the data, please test it.
I haven't deployed yet grafana.

**Comment by zonca** [comment]
forgot to say that I have temporarily deployed 1 medium master (as usual) and 1 medium worker (we usually use xlarge here) to save resources. So you cannot use the "Full node" option with 50 GB of RAM for now. @pibion is it ok?
When needed I can remove the medium worker and deploy a xlarge.

**Comment by zonca** [comment]
ok, grafana is back online (a newer version), see how to access (nothing changed) https://github.com/pibion/jupyterhub-deploy-kubernetes-jetstream-secrets#access-the-grafana-monitoring-dashboard

**Comment by zonca** [comment]
Something went wrong in the transfer of volumes, only these are saved:

* `zonca`, because I transferred it manually before doing all others in batch
* `rac101`, but it is in `attaching` status so it could be broken
* `mdlama`, but it is in `reserved` status so it could be broken

all the other user volumes are lost (the data volume is fine). @pibion sorry about this.

I think I know that went wrong, I didn't switch the reclaim policy before transfer, I did for my volume, but then though it was unnecessary and didn't do for the others. I'll write into the docs for next time...

What is tricky is that the kubernetes volumes resources are all there but the backend openstack volume is gone. so I have to clean up and delete all the kubernetes resources. except mine, `rac101` and `mdlama` that will try to fix when they login next time.
